apps/api/src/domain/models/shared/common-commands/provide-additional-credits-for-resource/additional-credits-provided-for-resource.event-handler.ts

```diff
@@ -0,0 +1,25 @@
+import { Inject } from '@nestjs/common';
+import { CoscradEventConsumer, ICoscradEventHandler } from '../../../../../domain/common';
+import { ContributionSummary } from '../../../user-management';
+import { QUERY_REPOSITORY_PROVIDER_TOKEN } from '../publish-resource/resource-published.event-handler';
+import { AdditionalCreditsProvidedForResource } from './additional-credits-provided-for-resource.event';
+
+export interface IQueryRepositoryForAttributable {
+    attribute(id: string, contributionSummary: ContributionSummary): Promise<void>;
+}
+
+interface IRepositoryProvider {
+    forResource(type: string): IQueryRepositoryForAttributable;
+}
+
+@CoscradEventConsumer('AdditionalCreditsProvidedForResource')
+export class AdditionalCreditsProvidedForResourceEventHandler implements ICoscradEventHandler {
+    constructor(
+        @Inject(QUERY_REPOSITORY_PROVIDER_TOKEN)
+        private readonly provider: IRepositoryProvider
+    ) {}
+
+    async handle(_event: AdditionalCreditsProvidedForResource): Promise<void> {
+        throw new Error('Method not implemented.');
```

Comment:
You need to get the repo and call the attribute method. You'll have to use the event payload to build the `ContributionSummary` for this one.

Reply:
```ts
    async handle({
        payload: {
            aggregateCompositeIdentifier: { type: resourceType, id: resourceId },
            contributionType,
            contributorIds,
        },
        meta: { dateCreated },
    }: AdditionalCreditsProvidedForResource): Promise<void> {
        const repo = this.provider.forResource(resourceType);

        if (!isFunction(repo?.attribute)) {
            return;
        }

        const contributionSummary: ContributionSummary = {
            contributorIds,
            statement: contributionType,
            type: contributionType,
            date: undefined,
            timestamp: dateCreated,
        };

        await repo.attribute(resourceId, contributionSummary);
    }
```

Reply:
Note that I am not 100% sure if that's what we want for the statement or if we want the word " by:" appended. We can experiment with manual testing to see how this should be or maybe take a snapshot in the handler test.